src/benchmark_mediation.py

```diff
@@ -1343,56 +1335,56 @@ def med_dml(
             )
         else:
             res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(x[train0], y[train0])
-        mu_t0_x[i] = res.predict(x[test])
-
-        # trimming
-        not_trimmed = (
-            (((1 - ptmx[i]) * ptx[i]) >= trim)
-            * ((1 - ptx[i]) >= trim)
-            * (ptx[i] >= trim)
-            * (((ptmx[i] * (1 - ptx[i]))) >= trim)
-        )
-        for var in var_name:
-            exec(f"{var}[i] = {var}[i][not_trimmed]")
-        nobs += np.sum(not_trimmed)
+        mu_t0_x[test] = res.predict(x[test])
+
+    # trimming
+    not_trimmed = (
+        (((1 - ptmx) * ptx) >= trim)
+        * ((1 - ptx) >= trim)
+        * (ptx >= trim)
+        * (((ptmx * (1 - ptx))) >= trim)
+    )
+    for var in var_name:
+        exec(f"{var} = {var}[not_trimmed]")
+    nobs = np.sum(not_trimmed)
 
     # score computing
     if normalized:
-        sumscore1 = [np.mean(_) for _ in (1 - tte) * ptmx / ((1 - ptmx) * ptx)]
-        sumscore2 = [np.mean(_) for _ in tte / ptx]
-        sumscore3 = [np.mean(_) for _ in (1 - tte) / (1 - ptx)]
-        sumscore4 = [np.mean(_) for _ in tte * (1 - ptmx) / (ptmx * (1 - ptx))]
-        y1m1 = (tte * (yte - mu_t1_x) / ptx) / sumscore2 + mu_t1_x
-        y0m0 = ((1 - tte) * (yte - mu_t0_x) / (1 - ptx)) / sumscore3 + mu_t0_x
+        sumscore1 = np.mean(t / ptx)
+        sumscore2 = np.mean((1 - t) / (1 - ptx))
+        sumscore3 = np.mean(t * (1 - ptmx) / (ptmx * (1 - ptx)))
+        sumscore4 = np.mean((1 - t) * ptmx / ((1 - ptmx) * ptx))
+        y1m1 = (t / ptx * (y - mu_t1_x)) / sumscore1 + mu_t1_x
+        y0m0 = ((1 - t) / (1 - ptx) * (y - mu_t0_x)) / sumscore2 + mu_t0_x
         y1m0 = (
```

Comment:
Are there tests that check whether this snippet computes the right thing ?
Such computation blocks should better be isolated into small functions bzw.

Reply:
There aren't any test, I just made sure it gives similar performances with and without normalization on several dataset. I can declare that in an issue about that